chain/db/delegation.go

```diff
@@ -0,0 +1,96 @@
+package db
+
+import (
+	"errors"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+func (b *BandDB) SetDelegation(
+	delegatorAddress sdk.AccAddress,
+	validatorAddress sdk.ValAddress,
+	shares string,
+) error {
+	return b.tx.Where(Delegation{
+		DelegatorAddress: delegatorAddress.String(),
+		ValidatorAddress: validatorAddress.String(),
+	}).
+		Assign(Delegation{Shares: shares}).
+		FirstOrCreate(&Delegation{}).Error
+}
+
+func (b *BandDB) RemoveDelegation(
+	delegatorAddress sdk.AccAddress,
+	validatorAddress sdk.ValAddress,
+) error {
+	return b.tx.Delete(Delegation{
+		DelegatorAddress: delegatorAddress.String(),
+		ValidatorAddress: validatorAddress.String(),
+	}).Error
+}
+
+func (b *BandDB) delegate(
```

Comment:
Can a validator delegate for himself? Does `SelfDelegation` need to increase?

Reply:
SelfDelegation is just a delegation record, not a variable in validator struct, logic query delegation by validator to compare with MinSelfDelegation

Reply:
Make sense.